src/Api.js

```diff
@@ -89,13 +92,68 @@ class Api {
 	}
 
 	/**
-	 * Get the WhoColor data for a given wiki page.
+	 * Get the WikiWho data for a given wiki page. Use onError and onResults to set return handlers.
 	 *
-	 * @param  {string} wikiUrl URL of the wiki page that we want to analyze.
-	 * @return {Promise}
+	 * @param {string} wikiUrl URL of the wiki page that we want to analyze.
 	 */
 	getData( wikiUrl ) {
-		return $.getJSON( this.getAjaxURL( wikiUrl ) );
+		$.getJSON( this.getAjaxURL( wikiUrl ) ).then( ( function ( result ) {
+			// Handle error.
+			if ( !result.success && result.info && this.errorHandler ) {
+				// Turn the English error message into a i18n message key.
+				// @TODO Remove this when the WikiWho API returns useful error codes.
+				let errorCode = result.info
+					.toLowerCase()
+					.replace( / /g, '-' )
+					.replace( /[^a-z-]/g, '' );
+				// Possible messages:
+				// eslint-disable-next-line
+				// ext-whowrotethat-requested-data-is-not-currently-available-in-wikiwho-database-it-will-be-available-soon
+				this.errorHandler( 'ext-whowrotethat-' + errorCode );
```

Comment:
Eh, I don't think we should do this. A couple of reasons:

- As I wrote about in the i18n, it results in extremely long keys
- In general, best practice is to avoid auto-generating message keys, since it's hard to fetch where they're used when you want to fix them. You're doing well by adding a comment with the key, but if we auto-generate we risk having a ton of generated keys eventually, which will make it unweildy.
- There are multiple cases where we might want to use the same message for several different errors. If we do it this way, we'll have to duplicate the translations for the translators and have redundant keys.
- If the text changes from the wikiwho API, our entire translation keys change, which is a mess for translators, worse than if only the message itself changes with the key remaining the same :\

Since we're already looking at the returning text, I think the better way to do this is to assign, ourselves, either error codes or error shorthands. You can still "normalize" the string we're getting so we don't mess with punctuation on those things, but simplify them to generic codes. For example, doing something like
```
if ( errorCode.toLower() === 'requested-data-is-not-currently-available-in-wikiwho-database-it-will-be-available-soon' ) {
 this.errorHandler( '100', 'wikiwho-generic' )
}
```

Or something like that. Then as a followup (another .then perhaps) we go with a switch that can easily group some of those together for the same message, or output a message for whatever is relevant.

Reply:
Good point. I think the code using Api can sort this out maybe. It feels like more of a UI matter.

Reply:
I think the UI should be the one picking what actual mw message to display, but the API class should simplify the errors into manageable codes/strings/whatever.

So the API doesn't have to deal with actual message keys, it can "just" output some base code -- the UI then has some `switch` statement based on what product tells us, that goes "error codes x, y get message key `blah`, error code z gets message key `foo`" and we can change whatever displays what in the UI -- but for that, the API needs to simplify them.

You're doing that in the next version anyways, but I think we might want to polish some of it once we implement the actual error message handling in the other ticket.